src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -1192,10 +1192,13 @@ class BookingService(
     else -> throw RuntimeException("Unknown premises type ${booking.premises::class.qualifiedName}")
   }
 
-  fun getCancelleableBookings(application: ApprovedPremisesApplicationEntity): List<BookingEntity> =
-    bookingRepository.findAllByApplication(application).filter { isCancellableCas1(it) }
+  fun getCancelleableCas1BookingsForUser(user: UserEntity, application: ApprovedPremisesApplicationEntity): List<BookingEntity> =
+    bookingRepository.findAllByApplication(application).filter { booking ->
+      isInCancellableStateCas1(booking) && userAccessService.userCanCancelBooking(user, booking)
+    }
 
-  private fun isCancellableCas1(booking: BookingEntity) = (booking.cancellation == null) && (booking.arrivals.isEmpty())
+  private fun isInCancellableStateCas1(booking: BookingEntity)
+    = (booking.cancellation == null) && (booking.arrivals.isEmpty())
```

Comment:
Not really part of this piece of work, but are we satisfied that "having no arrivals" is a sufficient guard against the possibility of cancelling a booking which has in fact been taken up?  Do we need to understand and document the process for restoring a booking which is in fact either i) currently under occupation or ii) was used in the past (e.g. in a ROTL plan)?

If we were to add an additional control to only permit cancellations of bookings which are in the future, I guess this is likely to mean than we would be unable to clean up crufty placements which ought to have been cancelled earlier but for "reasons" have remained uncancelled beyond their arrival date?

Reply:
"Do we need to understand and document the process for restoring a booking which is in fact either i) currently under occupation or ii) was used in the past (e.g. in a ROTL plan)?"

I'm not aware of ROTL plans so can't answer to that, but i do think we could have a case where an arrival isn't recorded (e..g they just forget) and in the meantime the booking is cancelled. In which case you may need the ability to 'restore' a booking. Maybe this is all edge case stuff that is out of scope for the current solution? Adam could probably address it best i'd imagine